src/Microsoft.ServiceHub.Framework/ServiceJsonRpcDescriptor.cs

```diff
@@ -279,23 +279,58 @@ public override RpcConnection ConstructRpcConnection(IDuplexPipe pipe)
 		return this.CreateConnection(jsonRpc);
 	}
 
-	/// <inheritdoc />
-	protected override ServiceRpcDescriptor Clone() => new ServiceJsonRpcDescriptor(this);
+	/// <summary>
+	/// Returns an instance of <see cref="ServiceJsonRpcDescriptor"/> that resembles this one,
+	/// but with <see cref="ExceptionStrategy"/>, <see cref="MultiplexingStreamOptions"/>, <see cref="JoinableTaskFactory"/> and <see cref="TraceSource"/> matching to passed in descriptor.
+	/// </summary>
+	/// <param name="copyFrom">The descriptor to copy settings from.</param>
+	/// <returns>A clone of this instance, with the properties changed or this same instance if the properties already match.</returns>
+	internal ServiceJsonRpcDescriptor WithSettingsFrom(ServiceJsonRpcDescriptor copyFrom)
+	{
+		if (this.ExceptionStrategy == copyFrom.ExceptionStrategy &&
+			this.MultiplexingStreamOptions == copyFrom.MultiplexingStreamOptions &&
+			this.MultiplexingStream == copyFrom.MultiplexingStream &&
+			this.JoinableTaskFactory == copyFrom.JoinableTaskFactory &&
+			this.TraceSource == copyFrom.TraceSource)
```

Comment:
I'm curious why these need to be copied to the wrapping descriptor. Shouldn't they only impact the methods _on_ the class, and if so, by delegating to the inner object, it will see the values for these properties?

If the outer object _also_ has these properties, and if the holder of the delegating wrapper were to _change_ one of those properties, do you expect it to change on the inner object too?
Perhaps one of the advantages of deriving the delegator class from `ServiceRpcDescriptor` instead (as I suggest in another comment) is that some or all these properties will vanish, leaving it far more clear that the properties that matter are on the inner object.

Reply:
They actually impact methods on the inner implementation and without applying these to the inner descriptor as well, we get a very wrong behavior.

For example, if one has a custom JsonRpcDescriptor that customizes CreateFormatter and we wrap it with a descriptor inheriting from DelegatingJsonRpcDescriptor (call it MyDescriptor) and override CreateJsonRpc wanting to combine behavior from both. 

We will pass our own instance of MyDescriptor and if someone calls WithMultiplexingStream() on that, we would want the stream options to be applied during CreateFormatter. However if we don't pass the settings here, that won't happen because inner descriptor was created without multiplexing stream options. The delegating descriptor class wouldn't be useful without these.

One of the unit tests specifically covers this scenario.